python/lsst/ts/externalscripts/maintel/warmup_hexapod.py

```diff
@@ -0,0 +1,265 @@
+              hexapod:
+                description: Which hexapod will move? (camera_hexapod/m2_hexapod)
+                type: string
+                enum: ["camera_hexapod", "m2_hexapod"]
+              axis:
+                description: Which axis will move? (x, y, z, u, v, w).
+                type: string
+                enum: ["x", "y", "z", "u", "v", "w"]
+                default: z
+              step_size:
+                description: >-
+                  The discrete step size in microns in which the hexapod will move. 
+                  This can be a number or an array of numbers.
+                anyOf:
+                  - type: number
+                  - type: array
+                default: 250
+              sleep_time:
+                description: The sleep time in seconds between movements movements.
+                anyOf:
+                  - type: number
+                  - type: array
+                default: 1
+              max_position:
+                description: Maximum position (absolute value) for the movements.
+                type: number
+                default: 13000
+            additionalProperties: false
+            """
+        return yaml.safe_load(yaml_schema)
+
+    async def configure(self, config):
+        """Configure the script.
+
+        Parameters
+        ----------
+        config: `types.SimpleNamespace`
+            Configuration data. See `get_schema` for information about data
+            structure.
+        """
+        self.log.debug(
+            f"Setting up configuration: \n"
+            f"  hexapod: {config.hexapod}\n"
+            f"  axis: {config.axis}\n"
+            f"  step_size: {config.step_size}\n"
+            f"  sleep_time: {config.sleep_time}\n"
+            f"  max_position: {config.max_position}\n"
+        )
+
+        self.config = config
+
+        # Converts the string to salindex
+        self.hexapod_name = config.hexapod
+        self.sal_index = getattr(SalIndex, self.hexapod_name.upper())
+        self.hexapod = getattr(self.mtcs.rem, f"mthexapod_{self.sal_index}")
+        self.log.debug(f"Using the {self.hexapod_name} with sal index {self.sal_index}")
+
+        # Assert step_size and sleep_time are compatible
+        if isinstance(self.config.step_size, (int, float)):
+            self.config.step_size = [self.config.step_size]
+
+        if isinstance(self.config.sleep_time, (int, float)):
+            self.config.sleep_time = [self.config.sleep_time]
+
+        if len(self.config.step_size) != len(self.config.sleep_time):
+            raise ValueError(
+                f"Expected `step_size` and `sleep_time` to have the same number of elements. "
+                f"Found len(step_size) ({len(config.step_size)}) different from "
+                f" len(sleep_time) = ({len(config.sleep_time)})."
+            )
+
+        # Relay the function used to execute the movement
+        if self.hexapod_name == "camera_hexapod":
```

Comment:
maybe do:

```
self.move_hexapod = self.mtcs.move_camera_hexapod if self.hexapod_name == "camera_hexapod" else self.mtcs.move_m2_hexapod
```

maybe use sal_index to compare though because then you don't need to rely on a hardcoded string here, e.g.;

```
self.move_hexapod = self.mtcs.move_camera_hexapod if self.hexapod_sal_index == SalIndex.CAMERA_HEXAPOD else self.mtcs.move_m2_hexapod
```